camel/models/cohere_model.py

```diff
@@ -0,0 +1,216 @@
+)
+
+try:
+    import os
+
+    if os.getenv("AGENTOPS_API_KEY") is not None:
+        from agentops import LLMEvent, record
+    else:
+        raise ImportError
+except (ImportError, AttributeError):
+    LLMEvent = None
+
+
+class CohereModel(BaseModelBackend):
+    r"""Cohere API in a unified BaseModelBackend interface."""
+
+    def __init__(
+        self,
+        model_type: ModelType,
+        model_config_dict: Dict[str, Any],
+        api_key: Optional[str] = None,
+        url: Optional[str] = None,
+        token_counter: Optional[BaseTokenCounter] = None,
+    ):
+        import cohere
+
+        super().__init__(
+            model_type, model_config_dict, api_key, token_counter=token_counter
+        )
+        self._api_key = api_key or os.environ.get("COHERE_API_KEY")
+        self._url = url or os.environ.get("COHERE_SERVER_URL")
+
+        self._client = cohere.Client(api_key=self._api_key)
+        self._token_counter: Optional[BaseTokenCounter] = None
+
+    def _to_openai_response(
+        self, response: 'NonStreamedChatResponse'
+    ) -> ChatCompletion:
+        unique_id = str(uuid.uuid4())
+
+        # Safely access nested attributes
+        def safe_get(obj, *keys):
+            for key in keys:
+                if isinstance(obj, dict):
+                    obj = obj.get(key)
+                else:
+                    obj = getattr(obj, key, None)
+                if obj is None:
+                    return None
+            return obj
+
+        obj = ChatCompletion.construct(
+            id=unique_id,
+            choices=[
+                dict(
+                    index=0,
+                    message={
+                        "role": "assistant",
+                        "content": response.text,
+                    },
+                    finish_reason=safe_get(response, 'finish_reason'),
+                )
+            ],
+            created=safe_get(response, 'meta', 'created_at'),
+            model=self.model_type.value,
+            object="chat.completion",
+            usage={
+                "prompt_tokens": safe_get(
+                    response, 'meta', 'billed_tokens', 'prompt_tokens'
+                )
+                or 0,
+                "completion_tokens": safe_get(
+                    response, 'meta', 'billed_tokens', 'completion_tokens'
+                )
+                or 0,
+                "total_tokens": (
+                    (
+                        safe_get(
+                            response, 'meta', 'billed_tokens', 'prompt_tokens'
+                        )
+                        or 0
+                    )
+                    + (
+                        safe_get(
+                            response,
+                            'meta',
+                            'billed_tokens',
+                            'completion_tokens',
+                        )
+                        or 0
+                    )
+                ),
+            },
```

Comment:
the path we get usage data seems not correct, it should from `response.meta.tokens.input_tokens` and `response.meta.tokens.output_tokens`